File: src/eqn/ast/EqnAstNodeDivide.java
package eqn.ast;

import eqn.parser.exception.EqnException;

public class EqnAstNodeDivide extends EqnAstNodeBinaryOperation {

    public EqnAstNodeDivide(EqnAstNode leftOperand, EqnAstNode rightOperand) throws EqnException {
        super("/", leftOperand, rightOperand, PrecedenceType.Multiplication);
    }

    @Override
    public double evaluate() throws EqnException {
        return left().evaluate() / right().evaluate();
    }
}

File: src/eqn/ast/EqnAstNodeDivide.kt
package eqn.ast

import eqn.parser.exception.EqnException

class EqnAstNodeDivide(leftOperand: EqnAstNode?, rightOperand: EqnAstNode?) : EqnAstNodeBinaryOperation("/", leftOperand, rightOperand, PrecedenceType.Multiplication, Type.Division) {
    @Throws(EqnException::class)
    override fun evaluate(): Double {
        return left().evaluate() / right().evaluate()
    }
}